akka-http-core/src/main/scala/akka/http/impl/engine/ws/Websocket.scala

```diff
@@ -172,13 +172,12 @@ private[http] object Websocket {
           }
         }
       })
-      val pullIn = () ⇒ pull(in)
+      val pullIn = () ⇒ tryPull(in)
```

Comment:
This case was not a double-pull, it was a pull-when-closed that happens because we unconditionally execute the pull even when the stage was completed while the emit was running.